packages/graphcache-rn-async-storage/src/makeAsyncStorage.ts

```diff
@@ -0,0 +1,114 @@
+import { StorageAdapter } from '@urql/exchange-graphcache';
+import AsyncStorage from '@react-native-async-storage/async-storage';
+import NetInfo from '@react-native-community/netinfo';
+
+export type StorageOptions = {
+  dataKey?: string;
+  metadataKey?: string;
+  maxAge?: number; // Number of days
+};
+
+let disconnect;
+
+export const makeAsyncStorage: (ops?: StorageOptions) => StorageAdapter = ({
+  dataKey = 'graphcache-data',
+  metadataKey = 'graphcache-metadata',
+  maxAge = 7,
+} = {}) => {
+  const todayDayStamp = Math.floor(
+    new Date().valueOf() / (1000 * 60 * 60 * 24)
+  );
+  const todayBatch = {};
+  const prefixCheck = new RegExp(`^${dataKey}_(\\d+)$`);
+
+  return {
+    /**
+     * On initial read, pull storage keys and see which match our storage key pattern.
+     * Any record that is expired should be removed from AsyncStorage.
+     * The rest of the batches should be merged into a cache in chrono order.
+     * Also hydrate todayBatch if we find it.
+     */
+    readData: async () => {
+      const cache = {};
+
+      try {
+        const persistedDayStamps = (await AsyncStorage.getAllKeys())
+          .reduce<number[]>((filtered, curr) => {
+            const match = curr.match(prefixCheck);
+            const ts = match ? match[1] : undefined;
+
+            if (ts) {
+              filtered.push(Number(ts));
+            }
+            return filtered;
+          }, [])
+          .sort();
+
+        // eslint-disable-next-line es5/no-for-of
+        for (const dayStamp of persistedDayStamps) {
```

Comment:
`for-of` may or may not be transpiled here, but I'd just axe it. Even if it isn't transpiling and works correctly, it may still be a bit of an inconsistency compared to the rest of the codebase across `urql`

Reply:
Actually I think this raises an underlying question of "why are we doing so many async calls anyway". I've updated the logic to store all the data against a single key instead.